packages/compass-smoke-tests/src/tests/read-only.ts

```diff
@@ -0,0 +1,57 @@
+import fs from 'node:fs';
+import { type SmokeTestsContext } from '../context';
+import { execute } from '../execute';
+import { getInstaller } from '../installers';
+import { createSandbox } from '../directories';
+import { getTestSubject } from '../test-subject';
+
+export async function testReadOnly(context: SmokeTestsContext) {
```

Comment:
This whole file is basically a copy/paste of time-to-first-query.ts. We can probably refactor it.